zkvm/src/signature/mod.rs

```diff
@@ -16,11 +16,59 @@ mod multikey;
 mod musig;
 mod signer;
 
-/// Verification key (aka "pubkey") is a wrapper type around a Ristretto point
+/// Verification key (aka "pubkey") is a wrapper type around a Ristretto point and it's compressed variation
 /// that lets the verifier to check the signature.
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
-pub struct VerificationKey(pub CompressedRistretto);
+pub struct VerificationKey {
+    point: RistrettoPoint,
+    // Available only if instantiated using `with_compressed`
+    // or if caller directly calls a method that returns a compressed point
+    compressed: Option<CompressedRistretto>,
```

Comment:
i'm thinking that we often have to use compressed point because we commit it somewhere - Keytree needs it to compute the additive factor, Multikey needs it to compute multiplicative factors. So let's simplify the logic and always store compressed point alongside decompressed one? And rename the field to reflect the fact that it's an optimization.

```suggestion
    precompressed: CompressedRistretto,
```

Then `From<RistrettoPoint>` would have to compress, but that's fine.

Reply:
With this, I think we can make most of the methods that fail with a compressedPoint non-failable; instead of passing the compressedPoint which could fail with `from_point`, we can pass the RistrettoPoint instead, since the compression is done regardless. Is this correct?